File: packages/autonavi_search_flutter/ios/Classes/AutonaviSearchPlugin.swift
import Flutter

public class AutonaviSearchPlugin: NSObject, FlutterPlugin {

    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(
            name: "plugins.autonavi.flutter/search",
            binaryMessenger: registrar.messenger()
        )
        let handler = SearchChannelHandler()
        registrar.addMethodCallDelegate(handler, channel: channel)
    }
}

File: packages/autonavi_search_flutter/android/src/main/kotlin/com/autonavi/search/flutter/AutonaviSearchPlugin.kt
package com.autonavi.search.flutter

import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodChannel

class AutonaviSearchPlugin : FlutterPlugin {

    private lateinit var channel: MethodChannel
    private var channelHandler: SearchChannelHandler? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channelHandler = SearchChannelHandler(binding.applicationContext)
        channel = MethodChannel(
            binding.binaryMessenger,
            "plugins.autonavi.flutter/search"
        )
        channel.setMethodCallHandler(channelHandler)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
        channelHandler = null
    }
}
